tests/ci/setup.sh

```diff
@@ -25,54 +27,78 @@ install_dependencies() {
+  sudo dpkg -i "${ROOT_DIR}/build/$HEROKUISH_PACKAGE_NAME" \
+    "${ROOT_DIR}/build/$PLUGN_PACKAGE_NAME" \
+    "${ROOT_DIR}/build/$SSHCOMMAND_PACKAGE_NAME" \
+    "${ROOT_DIR}/build/$SIGIL_PACKAGE_NAME" \
+    "${ROOT_DIR}/build/$PROCFILE_UTIL_PACKAGE_NAME"
+}
+
+build_dokku() {
+  echo "=====> build_dokku on CIRCLE_NODE_INDEX: $CIRCLE_NODE_INDEX"
+  "${ROOT_DIR}/contrib/release-dokku" build
+  docker build -t dokku/dokku:test .
+}
+
+run_dokku_container() {
+  echo "=====> run_dokku_container on CIRCLE_NODE_INDEX: $CIRCLE_NODE_INDEX"
+  docker run -d \
+    --env DOKKU_HOSTNAME=dokku.me \
+    --name dokku \
+    --publish 3022:22 \
+    --publish 22333:22 \
+    --publish 80:80 \
+    --publish 443:443 \
+    --volume /var/lib/dokku:/mnt/dokku \
+    --volume /var/run/docker.sock:/var/run/docker.sock \
+    dokku/dokku:test
+
+  check_container
+}
 
-  sudo dpkg -i "${ROOT_DIR}/build/$HEROKUISH_PACKAGE_NAME"
-  sudo dpkg -i "${ROOT_DIR}/build/$PLUGN_PACKAGE_NAME"
-  sudo dpkg -i "${ROOT_DIR}/build/$SSHCOMMAND_PACKAGE_NAME"
-  sudo dpkg -i "${ROOT_DIR}/build/$SIGIL_PACKAGE_NAME"
-  sudo dpkg -i "${ROOT_DIR}/build/$PROCFILE_UTIL_PACKAGE_NAME"
+check_container() {
+  echo "=====> check_container on CIRCLE_NODE_INDEX: $CIRCLE_NODE_INDEX"
+  local is_up
+  local cnt=0
+  while true; do
+    echo "$(date) [count: $cnt]: waiting for dokku startup"
+    is_up=$(
+      docker exec -ti dokku ps -ef | grep "/usr/sbin/sshd -D" >/dev/null 2>&1
+      echo $?
+    )
+    if [[ $is_up -eq 0 ]]; then
+      echo "" && docker logs dokku
+      break
+    fi
+    sleep 2
+    cnt=$((cnt + 1))
+  done
 }
 
 install_dokku() {
+  echo "=====> install_dokku on CIRCLE_NODE_INDEX: $CIRCLE_NODE_INDEX"
+
   if [[ "$FROM_SOURCE" == "true" ]]; then
     sudo -E CI=true make -e install
     return
   fi
 
-  "${ROOT_DIR}/contrib/release-dokku" build
+  build_dokku
+  run_dokku_container
 
-  echo "dokku dokku/hostname string dokku.me" | sudo debconf-set-selections
-  echo "dokku dokku/key_file string /root/.ssh/id_rsa.pub" | sudo debconf-set-selections
-  echo "dokku dokku/nginx_enable boolean true" | sudo debconf-set-selections
-  echo "dokku dokku/skip_key_file boolean true" | sudo debconf-set-selections
-  echo "dokku dokku/vhost_enable boolean true" | sudo debconf-set-selections
-  echo "dokku dokku/web_config boolean false" | sudo debconf-set-selections
-  sudo dpkg -i "$(cat "${ROOT_DIR}/build/deb-filename")"
+  sudo mkdir -p /usr/local/bin
+  sudo cp -fv contrib/dokku-docker-bin.sh /usr/local/bin/dokku
 }
 
 # shellcheck disable=SC2120
 setup_circle() {
   echo "=====> setup_circle on CIRCLE_NODE_INDEX: $CIRCLE_NODE_INDEX"
-  sudo -E CI=true make -e sshcommand
-  # need to add the dokku user to the docker group
-  sudo usermod -G docker dokku
-  [[ "$1" == "buildstack" ]] && BUILD_STACK=true make -e stack
 
   install_dependencies
   install_dokku
 
   sudo -E make -e setup-deploy-tests
-  bash --version
-  docker version
```

Comment:
Should we have this output from the built container?

Reply:
i don't know that's useful these days. we can always add it back if we find a need

Reply:
Fair. I don't find it useful, so we can leave it off for now.